cmd/image-builder/main.go

```diff
@@ -33,11 +33,46 @@ func cmdListImages(cmd *cobra.Command, args []string) error {
 		return err
 	}
 
-	opts := &cmdlineOpts{
-		out:     osStdout,
-		dataDir: dataDir,
+	return listImages(dataDir, output, filter)
+}
+
+func cmdManifest(cmd *cobra.Command, args []string) error {
+	dataDir, err := cmd.Flags().GetString("datadir")
+	if err != nil {
+		return err
+	}
+
+	distroStr := args[0]
+	imgTypeStr := args[1]
+	var archStr string
+	if len(args) > 2 {
+		archStr = args[2]
+	} else {
+		archStr = arch.Current().String()
 	}
-	return listImages(output, filter, opts)
+	// XXX: this is not symetric, ideally we would only pass
+	// either a single reporegistry from a datadir everywhere
+	// or we would pass datadir everywhere but with just
+	// datadir we cannot create test distros as they cannot be
+	// defined outside of code so manifestgen needs to take a
+	// reporegistry instead of just a datadir
+	res, err := getOneImage(dataDir, distroStr, imgTypeStr, archStr)
+	if err != nil {
+		return err
+	}
```

Comment:
I'm not sure I completely understand the comment here.
Why can't test distros be defined outside?

Reply:
Below is long and probably too much of-internal monologue explanation but you don't need to read it as I moved on from "test-distro" now and just use the real ones for testing. Different tradeoffs but easier to work with and arguably more realistic :)

---

Sorry, maybe I should remove this paragraph. The background is that I am trying to use the `test-distro-1` in my tests. And there are some steps for that with that and the distrofactory that I would love to simplify. 

Right now one needs to do:
1. distrofactory needs repositories to create distros, repositories are defined in json files
2. disrofactory also needs registered factory functions
3.  once distrofactory.Factory is created no new factories can be created
4. this means that for testing one needs to create a "distrofactory.NewTestDefault" (becuase (3))
5. it also means for testing one needs to put a custom datadir into the reporegistry with `test-distro-1.json` and empty fake architecture 

So tests that use test-distro-1 need to create a custom datadir and mock the distrofactory.NewDefault because one source of data is "datadir" and the other is registered factory functions in code. It would be nice to be able to just define a datadir and get my test distros but that would means we would have to have an (optional?) way to create distros from yaml (i.e. I would like to get rid of the need to mock distrofactory.NewDefault).

I guess the alternative is to just not use test-distro here and just use a "real" one which is easier to do as the default factory will work. I think I will try this.

Reply:
Fwiw, the paragraph is removed because test-distro is no loner used.

Reply:
Thanks for the explanation :)